application.ts

```diff
@@ -39,8 +42,8 @@ export class Application<S extends object = { [key: string]: any }> {
   }
 
   /** Register middleware to be used with the application. */
-  use(...middleware: Middleware<S, Context<S>>[]): this {
-    this._middleware.push(...middleware);
-    return this;
+  use<NS extends {} = S>(middleware: Middleware<NS, Context<NS>>) {
+    this._middleware.push(middleware);
+    return (this as unknown) as Application<NS extends S ? NS : (S & NS)>;
```

Comment:
I would prefer something like this:

```ts
  use<NS extends {} = S>(
    middleware: Middleware<NS, Context<NS>>
  ): Application<NS extends S ? NS : (S & NS)> {
    this._middleware.push(middleware);
    return this as unknown;
  }
```

Reply:
I agree it's better to explicitly declare return type, but we have to change `unknow` to `any` to make it  compile